IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/MonoLayerAir.mo

```diff
@@ -55,25 +66,42 @@ model MonoLayerAir
     else 1 "Correlations from Hollands et al. and Wright et al.";
 
 protected
-  final parameter Boolean ceiling=IDEAS.Utilities.Math.Functions.isAngle(inc,IDEAS.Types.Tilt.Ceiling)
+  final parameter Boolean ceiling=
+     IDEAS.Utilities.Math.Functions.isAngle(inc,IDEAS.Types.Tilt.Ceiling)
     "true if ceiling"
     annotation(Evaluate=true);
-  final parameter Boolean floor=IDEAS.Utilities.Math.Functions.isAngle(inc,IDEAS.Types.Tilt.Floor)
+  final parameter Boolean floor=
+     IDEAS.Utilities.Math.Functions.isAngle(inc,IDEAS.Types.Tilt.Floor)
     "true if floor"
     annotation(Evaluate=true);
-  final parameter Boolean vertical=IDEAS.Utilities.Math.Functions.isAngle(inc,IDEAS.Types.Tilt.Wall)
+  final parameter Boolean vertical=
+     IDEAS.Utilities.Math.Functions.isAngle(inc,IDEAS.Types.Tilt.Wall)
     annotation(Evaluate=true);
-  final parameter Real coeffRa=Modelica.Constants.g_n*beta*d^3/nu/alpha "Coefficient for evaluating less operations at run time";
-  Real Ra = max(1,(if linearise then abs(dT_nominal) else abs(port_a.T-port_b.T))*coeffRa);
+  final parameter Real coeffRa=Modelica.Constants.g_n*beta*d^3/nu/alpha
+  "Coefficient for evaluating less operations at run time";
+  Real Ra = max(1,(if linearise then abs(dT_nominal) else
+    abs(port_a.T-port_b.T))*coeffRa);
   Modelica.SIunits.CoefficientOfHeatTransfer h = Nu*k/d;
 
 public
   Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port_emb "Internal port"
     annotation (Placement(transformation(extent={{-10,90},{10,110}})));
+
 equation
   if not (ceiling or floor or vertical) then
-    assert(false, "Could not find suitable correlation for air cavity! Please change the inclination to wall, ceiling or floor or remove the air layer.",
-         level=AssertionLevel.warning);
+    assert(false, "Could not find suitable correlation for air cavity! Please 
+      change the inclination to wall, ceiling or floor or remove the air layer.",
+      level=AssertionLevel.warning);
+  end if;
+
+  if checkCoating then
+    assert((epsLw_a <> IDEAS.Buildings.Data.Constants.epsLw_glass)
+      or (epsLw_b <> IDEAS.Buildings.Data.Constants.epsLw_glass),
+      "In " + getInstanceName() + ": You are probably unintentionally simulating low 
+      performance glazing by not including a coating in the glazing record, 
+      to correct it, modify the longwave emissivity within the glazing system.
+      This check can be disabled using the parameter checkLowPerformanceGlazing.",
+      level=AssertionLevel.error);
```

Comment:
AssertionLevel should be .warning, no?

And shouldn't we put the assert in initial equations? Or maybe it does not matter?

Reply:
I'd throw an assert since warnings are often ignored and we have an option for disabling the assert.

I've had better experience with an assert in the equation section. It seems as if they are sometimes already evaluated during the translation while asserts in the initial equation section are not. This is an advantage since the error is returned earlier. It's hard to reproduce though..